meal.go

```diff
@@ -175,3 +178,144 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
 
 	return
 }
+
+var regexDateFromTitle *regexp.Regexp = regexp.MustCompile("(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D?")
+
+// 현재 시간을 기준으로 해당 주의 시간표를 추출한다.
+func getMealDataWithTime(client HttpClient, now time.Time) (week []MealData, err error) {
+	defer WhereInError(&err, "현재 주간 학식")
+
+	weekday := now.Weekday()
+	if weekday < time.Monday || weekday > time.Friday {
+		err = NotFoundError.CreateError(errors.New("주말에는 학식이 제공하지 않습니다."))
+		return
+	}
+
+	// 학식 게시판 목록을 가져온다.
+	ids, err := getMealID(client)
+	if err != nil {
+		return
+	}
+
+	// 시간대 설정
+	loc, _ := time.LoadLocation("Asia/Seoul")
+
+	for _, id := range ids {
+		// 올린 날짜에서 년도를 가져오고
+		year, _ := strconv.Atoi(strings.Split(id.Date, "-")[0])
+
+		// 제목에서 기간을 가져온다.
+		parsedDuration := regexDateFromTitle.FindStringSubmatch(id.Title)
+		if len(parsedDuration) != 5 {
+			continue
+		}
+
+		// 가져온 기간을 숫자 배열로 다시 변환한다.
+		numberDuration := []int{}
+		for _, plain := range parsedDuration[1:] {
+			num, _ := strconv.Atoi(plain)
+			numberDuration = append(numberDuration, num)
+		}
+
+		// 년도와 기간을 통해 시작일과 종료일을 Time 자료형으로 생성한다.
+		startDay := time.Date(year, time.Month(numberDuration[0]), numberDuration[1], 0, 0, 0, 0, loc)
+		endDay := time.Date(year, time.Month(numberDuration[2]), numberDuration[3], 23, 59, 59, 99, loc)
+
+		// 현재 시간이 시작일과 종료일 사이에 있는지 확인한다.
+		if endDay.After(now) && startDay.Before(now) {
+			// 그 주의 학식을 가져온다.
+			week, err = getMealDataWithID(client, id.ID)
+			if err != nil {
+				return
+			}
+
+			return
+		}
+	}
+
+	err = NotFoundError.CreateError(errors.New("해당 기간에 학식이 존재하지 않습니다."))
+	return
+}
+
+// 현재 시간을 기준으로 해당 요일의 시간표를 추출한다.
+func getMealDataWithWeekday(client HttpClient, now time.Time, weekday int) (day []MealData, err error) {
+	defer WhereInError(&err, "요일별 학식")
+
+	if weekday < 1 || weekday > 5 {
+		err = NotFoundError.CreateError(errors.New("주말에는 학식이 제공하지 않습니다."))
+		return
+	}
+
+	data, err := getMealDataWithTime(client, now)
+	if err != nil {
+		return
+	}
+
+	day = []MealData{
+		data[weekday-1],
+	}
+	return
+}
+
+type GetMealDataResult struct {
+	Data []MealData `json:"data"`
+}
+
+// godoc GetMealData
+// @Summary 학식 게시판에서 학식을 가져온다.
+// @Description MealData 배열인 data를 가진 구조체를 리턴받는다.
+// @Produce json
+// @Param id query int false "게시물 ID" 382
+// @Param day query int false "날짜 요일 (0~5)" 5
+// @Success 200 {object} GetMealDataResult
+// @Failure 400 {object} ErrorMessage
+// @Failure 404 {object} ErrorMessage
+// @Failure 502 {object} ErrorMessage
+// @Router /meal/get [get]
+func GetMealData(c *gin.Context) {
+	// GET 인수를 받는다. 기본값은 문자열 "0"
+	plainId := c.DefaultQuery("id", "0")
```

Comment:
Codacy found an issue: [var plainId should be plainID](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)